designer/server/src/main/scala/pl/touk/nussknacker/ui/process/deployment/CustomActionInvokerService.scala

```diff
@@ -53,17 +53,13 @@ class CustomActionInvokerServiceImpl(processRepository: FetchingProcessRepositor
               if (customAction.allowedStateStatusNames.contains(status.status.name)) {
                 manager.invokeCustomAction(actionReq, process.json)
               } else
-                actionError(CustomActionInvalidStatus(actionReq, status.status.name))
+                Future.failed(CustomActionInvalidStatus(actionReq, status.status.name))
```

Comment:
If throwing exception doesn't work, extract some error message for Future.failed.